tests/runtime/prelude/core/primitives/String.sk

```diff
@@ -15,9 +15,9 @@ native class .String uses Hashable, Show, Orderable {
   @may_alloc
   native static fun fromUtf8(bytes: readonly Array<UInt8>): String;
 
-  // Careful, this is O(n) (because of utf8)
-  @cpp_runtime
-  native private fun unsafe_get(x: Int): Char;
+  static fun fromGenerator(gen: () -> mutable Iterator<Char>): String {
```

Comment:
Ideally we could prevent needing more and more "fromFoo()" type constructor methods. For this case, it seems like `fromIterator()` would be sufficient - it works with iterators, and if you have a generator function you just have to call it first to get an iterator. The only reason I can see to add this exact variation is that we don't have list comprehension style syntax. I'd like something like

```javascript
String::fromIterator(...while (cond) yield blah)
```

cc @mjhostet @tnowacki

Reply:
This was just handy because a lot of the other functions were in the form:
```
gen: () -> mutable Iterator<Char> = () -> {
   ... stuff ...
};
String::fromChars(Array::fromIterator(gen()))
```
If you dislike it I can make the functions use the pattern directly...